sweep/sweeper.go

```diff
@@ -545,6 +558,11 @@ func (s *UtxoSweeper) collector(blockEpochs <-chan *chainntnfs.BlockEpoch) {
 				pendInput.listeners = append(
 					pendInput.listeners, input.resultChan,
 				)
+
+				if prevExclGroup != nil {
```

Comment:
Shouldn't this be only done on _confirmation_?

Reply:
It is only done on confirmation, but from the PoV of the `ChannelArbitrator/anchorResolver`, since the anchor input is re-offered by the resolver without an exclusive group and the resolvers only run after a commitment transaction has confirmed.

Reply:
Cool, so then this is actually an alternative to: https://github.com/lightningnetwork/lnd/pull/4860 right?